core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -310,12 +319,6 @@ void validate() {
       if (!containsMacro("nodePath")) {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(nodePath), "Node path cannot be empty.");
       }
-      if (!containsMacro("tableName")) {
-        Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName), "Table name cannot be empty.");
-      }
-      if (!containsMacro("tableExpiryPeriod")) {
-        Preconditions.checkArgument(tableExpiryPeriod != null, "Table expiry period cannot be empty.");
-      }
       if (!containsMacro("temporaryFolder")) {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(temporaryFolder), "Temporary folder cannot be empty.");
       }
```

Comment:
add a check to verify the tableExpiryPeriod is > 0 if it is not null and does not contain a macro.